composite/action.yml

```diff
@@ -238,6 +239,14 @@ runs:
         LOG_LEVEL: ${{ inputs.log_level }}
       shell: bash
 
+    - name: Save PIP packages cache
+      uses: actions/cache/save@v3
+      if: always()
```

Comment:
`if: success() || failure()`

`always()` includes `cancelled()` which might result in a partial broken cache

https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions